src/components/molecules/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,123 @@
+import { Box } from '@mui/material'
+import React, { createRef, RefObject, useEffect, useState } from 'react'
+import './carousel.scss'
+
+export type CarouselProps = {
+  elementWidth: number
+  elementHeight: number
+  selfRotate: boolean
+  elements: React.ReactNode[]
+}
+
+const defaultCarouselProps = {
+  elementWidth: 210,
+  elementHeight: 140,
+  selfRotate: true,
+  elements: [],
+}
+
+export const Carousel = (props: CarouselProps = defaultCarouselProps) => {
+  const [selectedCell, setSelectedCell] = useState(0)
+  const [intervalId, setIntervalId] = useState<NodeJS.Timer | null>(null)
+
+  const carouselRef: RefObject<HTMLDivElement> = createRef<HTMLDivElement>()
+  // Refs to move the elemenets
+  const elementRefs: RefObject<HTMLDivElement>[] = props.elements.map(() => createRef<HTMLDivElement>())
+
+  function getHelperData() {
```

Comment:
Can this logic moved to a hook? or makes no sense?

Reply:
It's doable, but I think that doesn't make sense, it is tied to the logic of the carousel, so you wont use it again on another comp.